numbers_test.go

```diff
@@ -170,40 +176,108 @@ func TestValidateNumber(t *testing.T) {
+	{"BE0472429986", "BE", "0472429986", true, "MS_UNAVAILABLE", &ErrServiceUnavailable},
+	{"BE0472429986", "BE", "0472429986", true, "MS_MAX_CONCURRENT_REQ", &ErrServiceUnavailable},
 }
 
-func TestValidateNumberFormat(t *testing.T) {
-	for _, test := range tests {
-		valid, err := ValidateNumberFormat(test.number)
-		if err != nil {
-			if err.Error() != ErrCountryNotFound.Error() {
-				panic(err)
+func TestLookupValidateNumberExistence(t *testing.T) {
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockViesService := mocks.NewMockViesServiceInterface(ctrl)
+
+	for _, test := range lookupTests {
+		if len(test.fullVatNumber) >= 3 {
+			expectedViesResponseBody := test.expectedResponse
+			if expectedViesResponseBody == "" {
+				expectedViesResponseBody = expectedViesResponse(test.vatNumber, test.countryCode, test.isValid)
+			}
+			expectedViesResponse := &http.Response{
+				Body:       io.NopCloser(bytes.NewBufferString(expectedViesResponseBody)),
+				StatusCode: http.StatusOK,
 			}
-		}
 
-		if test.valid != valid {
-			t.Errorf("Expected %v for %v, got %v", test.valid, test.number, valid)
+			mockViesService.EXPECT().
+				Lookup(expectedViesRequestEnvelope(test.vatNumber, test.countryCode)).
+				Return(expectedViesResponse, nil)
 		}
 
+		viesResponse, err := Lookup(test.fullVatNumber, mockViesService)
+		if err != nil {
+			if test.expectedError == nil {
+				t.Error(fmt.Errorf("unexpected error: %w", err))
+			} else if err != *test.expectedError {
+				t.Error(fmt.Errorf("Expected error: %w\nGot: %s", *test.expectedError, err.Error()))
+			}
+		} else {
+			if viesResponse == nil {
+				t.Error("expected a ViesResponse, none received")
+			}
+			if test.isValid && !viesResponse.Valid {
+				t.Error(fmt.Sprintf("expected %s to be a valid VAT number.", test.fullVatNumber))
+			}
+			if !test.isValid && viesResponse.Valid {
+				t.Error(fmt.Sprintf("expected %s to not be a valid VAT number.", test.fullVatNumber))
+			}
+		}
 	}
 }
 
-func TestValidateNumberExistence(t *testing.T) {
-	valid, _ := ValidateNumberExistence("BE0472429986")
-	if !valid {
-		t.Error("Expected BE0472429986 to be a valid VAT number.")
-	}
+func expectedViesRequestEnvelope(vatNumber string, countryCode string) string {
+	return fmt.Sprintf(
+		`<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+<soapenv:Header/>
+<soapenv:Body>
+  <checkVat xmlns="urn:ec.europa.eu:taxud:vies:services:checkVat:types">
+	<countryCode>%s</countryCode>
+	<vatNumber>%s</vatNumber>
+  </checkVat>
+</soapenv:Body>
+</soapenv:Envelope>`,
+		vatNumber,
+		countryCode,
+	)
+}
 
-	valid, _ = ValidateNumberExistence("NL123456789B01")
-	if valid {
-		t.Error("Expected NL123456789B01 to not be a valid VAT number.")
+func expectedViesResponse(vatNumber string, countryCode string, isValid bool) string {
+	isValidString := "true"
+	if !isValid {
+		isValidString = "false"
 	}
```

Comment:
Could use `strconv.FormatBool(isValid)` here?

Reply:
Cool, didn't know about that. Looks like it works